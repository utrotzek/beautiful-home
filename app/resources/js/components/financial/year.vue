<template>
    <div>
        <button 
                class="btn btn-default text-left decrease-year"
                @click="decreaseYear()"
        >
            <span class="fa fa-minus-circle"></span>
        </button> 
        
        <span class="year-label">{{ currentYear }}</span>

        <button
                class="btn btn-default text-left increase-year"
                @click="increaseYear()"
        >
            <span class="fa fa-plus-circle"></span>
        </button>
    </div>
</template>

<script>
    export default {
        name: "year.vue",
        props: {
            year: {
                default:  2019,
                type: Number,
                required: true
            }
        },
        data: function () {
            return {
                currentYear: this.year
            }
        },
        methods: {
            increaseYear: function (){
                this.currentYear++;
                this.$emit("update-year", this.currentYear);
            },
            decreaseYear: function (){
                this.currentYear--;
                this.$emit("update-year", this.currentYear);
            }
        }
    }
</script>

<style scoped>
    .year-label {
        font-size: 140%;
    }
</style>