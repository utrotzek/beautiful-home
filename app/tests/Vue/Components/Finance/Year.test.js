import { mount } from "@vue/test-utils";
import year from "../../../../resources/js/components/Finance/Year.vue";

describe("year", () => {
    let wrapper = null;

    beforeEach(() => {
        wrapper = mount(
            year,
            {
                propsData: {
                    year: 2019
                }
            }
        );
    });

    it("can display the default templateCrud", () => {
        expect(wrapper.find(".templateCrud-label").text()).toEqual("2019");
    });

    it("can deacrease and increase templateCrud", () => {
        wrapper.find(".decrease-templateCrud").trigger("click");
        expect(wrapper.emitted("update-year")[0]).toEqual([2018]);
        wrapper.find(".decrease-templateCrud").trigger("click");
        expect(wrapper.emitted("update-year")[1]).toEqual([2017]);

        wrapper.find(".increase-templateCrud").trigger("click");
        expect(wrapper.emitted("update-year")[2]).toEqual([2018]);
        wrapper.find(".increase-templateCrud").trigger("click");
        expect(wrapper.emitted("update-year")[3]).toEqual([2019]);

    });

    function expectUpdatedValue(value){
    }
});
