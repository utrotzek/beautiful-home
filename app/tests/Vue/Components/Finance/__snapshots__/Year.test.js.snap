// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`year renders properly 1`] = `
<div>
  <button
    class="btn btn-default text-left decrease-year"
  >
    <span
      class="fa fa-minus-circle"
    />
  </button>
   
  <span
    class="year-label"
  >
    2019
  </span>
   
  <button
    class="btn btn-default text-left increase-year"
  >
    <span
      class="fa fa-plus-circle"
    />
  </button>
</div>
`;
